doc/man3/X509_LOOKUP.pod

```diff
@@ -0,0 +1,182 @@
+to enable lookup in that store.
+
+X509_LOOKUP_new() creates a new B<X509_LOOKUP> with the implementation
+in I<method>.
+They can also be created by calling L<X509_STORE_add_lookup(3)>, which
+will associate a B<X509_STORE> with the lookup mechanism.
+
+X509_LOOKUP_init() initializes the given B<X509_LOOKUP>, after which
+it can be put to work.
+
+X509_LOOKUP_shutdown() tears down the given B<X509_LOOKUP>.
+
+X509_LOOKUP_free() destructs the given B<X509_LOOKUP>.
+
+X509_LOOKUP_set_method_data() and X509_LOOKUP_get_method_data()
+associates and retrieves a pointer to application data to and from the
+given B<X509_LOOKUP>.
+
+X509_LOOKUP_ctrl() is used to set or get additional data to or from a
+B<X509_LOOKUP> structure or its associated L<X509_LOOKUP_METHOD(3)>.
+The exact information to be set or received is determined by the
+I<cmd> number, and the various arguments I<argc>, I<argl> and I<*ret>
+holds the data being passed.  The I<cmd>s known to OpenSSL are
+discussed in more depth in L</Control Commands>.
+
+X509_LOOKUP_load_file() passes a filename to be loaded immediately
+into the associated B<X509_STORE>.  I<type> indicates what type of
+object is expected.
+This can only be used with a lookup using the implementation
+L<X509_LOOKUP_file(3)>.
+
+X509_LOOKUP_add_dir() passes a directory specification from which
+certificates and CRLs are loaded on demand into the associated
+B<X509_STORE>.  I<type> indicates what type of object are expected.
+This can only be used with a lookup using the implementation
+L<X509_LOOKUP_hash_dir(3)>.
+
+X509_LOOKUP_add_store() passes a URI for a directory-like structure
+from which containers with certificates and CRLs are loaded on demand
+into the associated B<X509_STORE>.  X509_LOOKUP_load_store() passes a
+URI for a single container from which certificates and CRLs are
+immediately loaded into the associated B<X509_STORE>.  These functions
+can only be used with a lookup using the implementation
+L<X509_LOOKUP_store(3)>.
+
+X509_LOOKUP_load_file(), X509_LOOKUP_add_dir(),
+X509_LOOKUP_add_store(), and X509_LOOKUP_load_store() are implemented
+as macros that use X509_LOOKUP_ctrl().
+
+X509_LOOKUP_by_subject(), X509_LOOKUP_by_issuer_serial(),
+X509_LOOKUP_by_fingerprint(), and X509_LOOKUP_by_alias() look up
+certificates and CRLs in the L<X509_STORE(3)> associated with the
+B<X509_LOOKUP> using different criteria, where the looked up object is
+stored in I<ret>.  Some of the underlying B<X509_LOOKUP_METHOD>s will
+also cache objects matching the criteria in the associated
+B<X509_STORE>, which makes it possible to handle cases where the
+criteria have more than one hit.
+
+=head2 Control Commands
+
+The B<X509_LOOKUP_METHOD>s built into OpenSSL recognise the following
+X509_LOOKUP_ctrl() functions:
+
+=over 4
+
+=item B<X509_L_FILE_LOAD>
+
+This is the command that X509_LOOKUP_load_file() uses.  The filename
+is passed in B<argc>, and the type in <argl>.
```

Comment:
Linebreak after uses.?

Reply:
I believe we have a more lax attitude regarding that...  as long as we don't start bickering over one or two spaces again.

... but sure, why not

Reply:
Not because of the period-space-space stuff, but because the next sentence deserves to be on its own line.

Reply:
Oh, so you wanted an empty line in between? (pod translators process a paragraph at a time and do not respect line breaks unless in verbatim mode)

Reply:
Good point. I'll take a look at the final manual page before answering your question...

Reply:
Maybe better leave it as it is. The output looks very ragged with all the newlines.

But I noticed that in some locations you need to replace

     <argl>   ->  B<argl>

Reply:
You mean `I<argl>`

Reply:
Why is it `B<argc>` and `I<argl>`?

Reply:
'cause `B<argc>` is incorrect. I copied and pasted from somewhere and was sloppy, it seems.